File: translations/codenet/uncommented/C++2J/codellama/Code/s107204188.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int t = sc.nextInt();
        int minA = (int)Math.pow(10,9) + 1;
        int num = 0;
        int f = 0;
        ArrayList<Integer> A = new ArrayList<Integer>();
        for(int i = 0; i < n; i++){
            A.add(sc.nextInt());
        }
        for(int i = 0; i < n; i++){
            if(minA > A.get(i)){
                minA = A.get(i);
            }else{
                if(f == A.get(i) - minA)num++;
                else if(f < A.get(i) - minA){
                    num = 1;
                    f = A.get(i) - minA;
                }
            }
        }
        System.out.println(num);
    }
}

File: translations/codenet/uncommented/C++2Go/codellama/Code/s107204188.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var minA int = int(math.Pow(10, 9) + 1)
	var num int = 0

	var n, t, f int = 0, 0, 0
	fmt.Scan(&n, &t)
	A := make([]int, n)

	for i := 0; i < n; i++ {
		fmt.Scan(&A[i])
	}

	for i := 0; i < n; i++ {
		if minA > A[i] {
			minA = A[i]
		} else {

			if f == A[i]-minA {
				num++
			} else if f < A[i]-minA {
				num = 1
				f = A[i] - minA
			}

		}
	}

	fmt.Println(num)

}

